scripts/docs_config/introduction.rst

```diff
@@ -0,0 +1,102 @@
+==============
+ Introduction
+==============
+
+The amount of data that needs to be processed by modern workloads is continuously 
+growing. To address the increasing demand, memory subsystem of modern server 
+platforms is becoming heterogeneous. For example, High-Bandwidth Memory (HBM) 
+addresses throughput needs; the CXL protocol closes the capacity gap and tends 
+to improve memory utilization by memory pooling capabilities. Beyond CPU use 
+cases, there are GPU accelerators with their own memory on board. 
+
+Modern heterogeneous memory platforms present a range of opportunities. At the 
+same time, they introduce new challenges that could require software updates to 
+fully utilize the HW features. There are two main problems that modern 
+applications need to deal with. The first one is appropriate data placement and 
+data migration between different types of memory. The second one is how SW 
+should leverage different memory topologies. 
+
+All applications can be divided into two big groups: enlightened and 
+unenlightened. Enlightened applications explicitly manage data allocation 
+distribution among memory tiers and further data migration. Unenlightened 
+applications do not require any code modifications and rely on underlying 
+infrastructure which is in turn enlightened. An underlying infrastructure 
+refers to not only the OS with various memory tiering solutions to migrate 
+memory pages between tiers, but also middleware: frameworks and libraries. 
+
+==============
+ Architecture
+==============
+
+The Unified Memory Framework (`UMF`_) is a library for constructing allocators 
+and memory pools. It also contains broadly useful abstractions and utilities 
+for memory management. UMF allows users to manage multiple memory pools 
+characterized by different attributes, allowing certain allocation types to be 
+isolated from others and allocated using different hardware resources as 
+required. 
+
+A memory pool is a combination of a pool allocator and one or more memory 
+targets accessed by memory providers along with their properties and allocation 
+policies. Specifically, a memory provider is responsible for coarse-grained 
```

Comment:
I think the current statement is misleading I think when we describe a pool we should not say anything about memory targets. A pool is a combination of a single pool allocator and a single memory provider. And memory provider itself could be instantiated from memspaces/memory targets but it is not mandatory. I can create a memory provider instance using low-level explicit API without using memspaces.
I would simplify this to `A memory pool is a combination of a pool allocator instance and a memory provider instance along with their properties and allocation 
policies.`

Reply:
done